### PR TITLE
Lazer Replay Data

### PR DESCRIPTION
**Pull request by KommToby**
Worked on this overnight with a friend for an osu collab series we do.
Seems to work with lazer .osr files, with EOF checks for stable plays as they don't contain any ScoreInfo elements.
Minor changes still need to be done, including ScoreID definitions, and some more testing would be handy.

Thought I'd open the PR so you could take a look, and if you wanted to go forwards with parsing full lazer replays.

### CODE REVIEW

**Comment by KommToby** [comment]
Marking this as a draft as theres still some improvements that should be made before merging.

**Comment by KommToby** [comment]
After review and testing with various replay files, including the three already in the original repo, it seems like this is good to go.
Should also be compatible with current version of danser-go as well

File: replay.go [hunk 1]
```diff
@@ -113,17 +114,34 @@ func ParseReplay(file []byte) (r *Replay, err error) {
 		}
 	}
 
-	if b.Len() == 8 {
-		if r.ScoreID, err = rInt64(b); err != nil {
-			return nil, fmt.Errorf("reading ScoreID: %s", err)
+	if r.ScoreID, err = rInt64(b); err != nil {
```
Comment (Wieku) [inline replay.go]: Older stable replays had a 32bit score id so there should be a fallback for it that was removed.

**Review by Wieku — CHANGES_REQUESTED** [review]

File: replay.go (hunk 1 above)
Comment (KommToby) [reply replay.go]: I'll fix this when I'm back from my trip, cheers

**Comment by KommToby** [comment]
I forgor to re-request

**Comment by Wieku** [comment]
I had to fix structs, but it's looking good now I think

**Pull request merged**